src/components/FormComponent/FormComponent.js

```diff
@@ -0,0 +1,77 @@
+import React, { Component } from 'react';
+import InputFieldComponent from '../InputFieldComponent/InputFieldComponent';
+import "./FormComponent.css";
+
+class FormComponent extends Component {
+	constructor() {
+		super();
+		this.state = {
+			checkbox: false,
+			email: '',
+			firstName: '',
+			lastName: '',
+			password: '',
+		}	
+	}
+
+	handleInput = (e) => {
+		const target = e.target;
+		const value = target.type === 'checkbox' ? target.checked : target.value;
```

Comment:
It's better to separate functionality for checkbox from other inputs; Even though it's an 'input', it still functions in completely different manner